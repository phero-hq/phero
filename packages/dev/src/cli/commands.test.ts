import {
  parseClientCommand,
  parsePheroCommand,
  parseServerCommand,
} from "./commands"

describe.skip("parseServerCommand", () => {
  function run(cmd: string) {
    return parseServerCommand(cmd.split(" ").splice(1))
  }

  function err(cmd: string) {
    return () => run(cmd)
  }

  describe("common", () => {
    test("cmd:   'phero-server --version'", () => {
      expect(run("phero-server --version")).toMatchSnapshot()
    })

    test("cmd:   'phero-server -v'", () => {
      expect(run("phero-server -v")).toMatchSnapshot()
    })

    test("cmd:   'phero-server --help'", () => {
      expect(run("phero-server --help")).toMatchSnapshot()
    })

    test("cmd:   'phero-server -h'", () => {
      expect(run("phero-server -h")).toMatchSnapshot()
    })

    test("cmd:   'phero-server nothing'", () => {
      expect(err("phero-server nothing")).toThrow()
    })
  })

  describe("serve", () => {
    test("cmd:   'phero-server serve --help'", () => {
      expect(run("phero-server serve --help")).toMatchSnapshot()
    })

    test("cmd:   'phero-server serve -h'", () => {
      expect(run("phero-server serve -h")).toMatchSnapshot()
    })

    test("cmd:   'phero-server serve'", () => {
      expect(run("phero-server serve")).toMatchSnapshot()
    })

    test("cmd:   'phero-server serve --verbose'", () => {
      expect(run("phero-server serve --verbose")).toMatchSnapshot()
    })

    test("cmd:   'phero-server serve --port 1234'", () => {
      expect(run("phero-server serve --port 1234")).toMatchSnapshot()
    })

    test("cmd:   'phero-server serve -p 1234'", () => {
      expect(run("phero-server serve -p 1234")).toMatchSnapshot()
    })

    test("cmd:   'phero-server serve -p 1234 --verbose'", () => {
      expect(run("phero-server serve -p 1234 --verbose")).toMatchSnapshot()
    })
  })

  describe("build", () => {
    test("cmd:   'phero-server build --help'", () => {
      expect(run("phero-server build --help")).toMatchSnapshot()
    })

    test("cmd:   'phero-server build -h'", () => {
      expect(run("phero-server build -h")).toMatchSnapshot()
    })

    test("cmd:   'phero-server build'", () => {
      expect(run("phero-server build")).toMatchSnapshot()
    })

    test("cmd:   'phero-server build --verbose'", () => {
      expect(run("phero-server build --verbose")).toMatchSnapshot()
    })
  })

  describe("export", () => {
    test("cmd:   'phero-server export --help'", () => {
      expect(run("phero-server export --help")).toMatchSnapshot()
    })

    test("cmd:   'phero-server export -h'", () => {
      expect(run("phero-server export -h")).toMatchSnapshot()
    })

    test("cmd:   'phero-server export'", () => {
      expect(run("phero-server export")).toMatchSnapshot()
    })

    test("cmd:   'phero-server export --verbose'", () => {
      expect(run("phero-server export --verbose")).toMatchSnapshot()
    })
  })
})

describe.skip("parseClientCommand", () => {
  function run(cmd: string) {
    return parseClientCommand(cmd.split(" ").splice(1))
  }

  function err(cmd: string) {
    return () => run(cmd)
  }

  describe("common", () => {
    test("cmd:   'phero-client --version'", () => {
      expect(run("phero-client --version")).toMatchSnapshot()
    })

    test("cmd:   'phero-client -v'", () => {
      expect(run("phero-client -v")).toMatchSnapshot()
    })

    test("cmd:   'phero-client --help'", () => {
      expect(run("phero-client --help")).toMatchSnapshot()
    })

    test("cmd:   'phero-client -h'", () => {
      expect(run("phero-client -h")).toMatchSnapshot()
    })

    test("cmd:   'phero-client nothing'", () => {
      expect(err("phero-client nothing")).toThrow()
    })
  })

  describe("watch", () => {
    test("cmd:   'phero-client watch --help'", () => {
      expect(run("phero-client watch --help")).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch -h'", () => {
      expect(run("phero-client watch -h")).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch'", () => {
      expect(run("phero-client watch")).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch http://localhost:1234'", () => {
      expect(run("phero-client watch http://localhost:1234")).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch --verbose'", () => {
      expect(run("phero-client watch --verbose")).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch --port 1234'", () => {
      expect(run("phero-client watch --port 1234")).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch -p 1234'", () => {
      expect(run("phero-client watch -p 1234")).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch http://localhost:1234 --verbose --port 4321'", () => {
      expect(
        run("phero-client watch http://localhost:1234 --verbose --port 4321"),
      ).toMatchSnapshot()
    })

    test("cmd:   'phero-client watch ../api'", () => {
      expect(err("phero-client watch ../api")).toThrow()
    })
  })

  describe("build", () => {
    test("cmd:   'phero-client build --help'", () => {
      expect(run("phero-client build --help")).toMatchSnapshot()
    })

    test("cmd:   'phero-client build -h'", () => {
      expect(run("phero-client build -h")).toMatchSnapshot()
    })

    test("cmd:   'phero-client build'", () => {
      expect(run("phero-client build")).toMatchSnapshot()
    })

    test("cmd:   'phero-client build http://localhost:1234'", () => {
      expect(run("phero-client build http://localhost:1234")).toMatchSnapshot()
    })

    test("cmd:   'phero-client build --verbose'", () => {
      expect(run("phero-client build --verbose")).toMatchSnapshot()
    })

    test("cmd:   'phero-client build http://localhost:1234 --verbose'", () => {
      expect(
        run("phero-client build http://localhost:1234 --verbose"),
      ).toMatchSnapshot()
    })
  })
})

describe.skip("parsePheroCommand", () => {
  function run(cmd: string) {
    return parsePheroCommand(cmd.split(" ").splice(1))
  }

  function err(cmd: string) {
    return () => run(cmd)
  }

  describe("common", () => {
    test("cmd:   'phero --version'", () => {
      expect(run("phero --version")).toMatchSnapshot()
    })

    test("cmd:   'phero -v'", () => {
      expect(run("phero -v")).toMatchSnapshot()
    })

    test("cmd:   'phero --help'", () => {
      expect(run("phero --help")).toMatchSnapshot()
    })

    test("cmd:   'phero -h'", () => {
      expect(run("phero -h")).toMatchSnapshot()
    })

    test("cmd:   'phero nothing'", () => {
      expect(err("phero nothing")).toThrow()
    })
  })

  describe("dev-env", () => {
    test("cmd:   'phero'", () => {
      expect(run("phero")).toMatchSnapshot()
    })
  })

  describe("client", () => {
    test("cmd:   'phero client watch'", () => {
      expect(run("phero client watch")).toMatchSnapshot()
    })

    test("cmd:   'phero client watch --port 1234'", () => {
      expect(run("phero client watch --port 1234")).toMatchSnapshot()
    })
  })

  describe("server", () => {
    test("cmd:   'phero server serve'", () => {
      expect(run("phero server serve")).toMatchSnapshot()
    })

    test("cmd:   'phero server serve --port 1234'", () => {
      expect(run("phero server serve --port 1234")).toMatchSnapshot()
    })
  })
})
