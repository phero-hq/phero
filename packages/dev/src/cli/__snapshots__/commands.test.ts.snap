// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseClientCommand build cmd:   'phero-client build --help' 1`] = `
{
  "command": "build",
  "name": "help",
}
`;

exports[`parseClientCommand build cmd:   'phero-client build --verbose' 1`] = `
{
  "name": "build",
  "server": {
    "url": "http://localhost:3030",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand build cmd:   'phero-client build -h' 1`] = `
{
  "command": "build",
  "name": "help",
}
`;

exports[`parseClientCommand build cmd:   'phero-client build http://localhost:1234 --verbose' 1`] = `
{
  "name": "build",
  "server": {
    "url": "http://localhost:1234",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand build cmd:   'phero-client build http://localhost:1234' 1`] = `
{
  "name": "build",
  "server": {
    "url": "http://localhost:1234",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand build cmd:   'phero-client build' 1`] = `
{
  "name": "build",
  "server": {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand common cmd:   'phero-client --help' 1`] = `
{
  "command": undefined,
  "name": "help",
}
`;

exports[`parseClientCommand common cmd:   'phero-client --version' 1`] = `
{
  "name": "version",
}
`;

exports[`parseClientCommand common cmd:   'phero-client -h' 1`] = `
{
  "command": undefined,
  "name": "help",
}
`;

exports[`parseClientCommand common cmd:   'phero-client -v' 1`] = `
{
  "name": "version",
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch --help' 1`] = `
{
  "command": "watch",
  "name": "help",
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch --port 1234' 1`] = `
{
  "name": "watch",
  "port": 1234,
  "server": {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch --verbose' 1`] = `
{
  "name": "watch",
  "port": 4040,
  "server": {
    "url": "http://localhost:3030",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch -h' 1`] = `
{
  "command": "watch",
  "name": "help",
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch -p 1234' 1`] = `
{
  "name": "watch",
  "port": 1234,
  "server": {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch http://localhost:1234 --verbose --port 4321' 1`] = `
{
  "name": "watch",
  "port": 4321,
  "server": {
    "url": "http://localhost:1234",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch http://localhost:1234' 1`] = `
{
  "name": "watch",
  "port": 4040,
  "server": {
    "url": "http://localhost:1234",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand watch cmd:   'phero-client watch' 1`] = `
{
  "name": "watch",
  "port": 4040,
  "server": {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parsePheroCommand client cmd:   'phero client watch --port 1234' 1`] = `
{
  "argv": [
    "watch",
    "--port",
    "1234",
  ],
  "name": "client",
}
`;

exports[`parsePheroCommand client cmd:   'phero client watch' 1`] = `
{
  "argv": [
    "watch",
  ],
  "name": "client",
}
`;

exports[`parsePheroCommand common cmd:   'phero --help' 1`] = `
{
  "command": undefined,
  "name": "help",
}
`;

exports[`parsePheroCommand common cmd:   'phero --version' 1`] = `
{
  "name": "version",
}
`;

exports[`parsePheroCommand common cmd:   'phero -h' 1`] = `
{
  "command": undefined,
  "name": "help",
}
`;

exports[`parsePheroCommand common cmd:   'phero -v' 1`] = `
{
  "name": "version",
}
`;

exports[`parsePheroCommand dev-env cmd:   'phero' 1`] = `
{
  "name": "dev-env",
  "verbose": false,
}
`;

exports[`parsePheroCommand server cmd:   'phero server serve --port 1234' 1`] = `
{
  "argv": [
    "serve",
    "--port",
    "1234",
  ],
  "name": "server",
}
`;

exports[`parsePheroCommand server cmd:   'phero server serve' 1`] = `
{
  "argv": [
    "serve",
  ],
  "name": "server",
}
`;

exports[`parseServerCommand build cmd:   'phero-server build --help' 1`] = `
{
  "command": "build",
  "name": "help",
}
`;

exports[`parseServerCommand build cmd:   'phero-server build --verbose' 1`] = `
{
  "name": "build",
  "verbose": true,
}
`;

exports[`parseServerCommand build cmd:   'phero-server build -h' 1`] = `
{
  "command": "build",
  "name": "help",
}
`;

exports[`parseServerCommand build cmd:   'phero-server build' 1`] = `
{
  "name": "build",
  "verbose": false,
}
`;

exports[`parseServerCommand common cmd:   'phero-server --help' 1`] = `
{
  "command": undefined,
  "name": "help",
}
`;

exports[`parseServerCommand common cmd:   'phero-server --version' 1`] = `
{
  "name": "version",
}
`;

exports[`parseServerCommand common cmd:   'phero-server -h' 1`] = `
{
  "command": undefined,
  "name": "help",
}
`;

exports[`parseServerCommand common cmd:   'phero-server -v' 1`] = `
{
  "name": "version",
}
`;

exports[`parseServerCommand export cmd:   'phero-server export --flavor gcloud-functions' 1`] = `
{
  "flavor": "gcloud-functions",
  "name": "export",
  "verbose": false,
}
`;

exports[`parseServerCommand export cmd:   'phero-server export --flavor nodejs --verbose' 1`] = `
{
  "flavor": "nodejs",
  "name": "export",
  "verbose": true,
}
`;

exports[`parseServerCommand export cmd:   'phero-server export --flavor nodejs' 1`] = `
{
  "flavor": "nodejs",
  "name": "export",
  "verbose": false,
}
`;

exports[`parseServerCommand export cmd:   'phero-server export --flavor vercel' 1`] = `
{
  "flavor": "vercel",
  "name": "export",
  "verbose": false,
}
`;

exports[`parseServerCommand export cmd:   'phero-server export --help' 1`] = `
{
  "command": "export",
  "name": "help",
}
`;

exports[`parseServerCommand export cmd:   'phero-server export -h' 1`] = `
{
  "command": "export",
  "name": "help",
}
`;

exports[`parseServerCommand serve cmd:   'phero-server serve --help' 1`] = `
{
  "command": "serve",
  "name": "help",
}
`;

exports[`parseServerCommand serve cmd:   'phero-server serve --port 1234' 1`] = `
{
  "name": "serve",
  "port": 1234,
  "verbose": false,
}
`;

exports[`parseServerCommand serve cmd:   'phero-server serve --verbose' 1`] = `
{
  "name": "serve",
  "port": 3030,
  "verbose": true,
}
`;

exports[`parseServerCommand serve cmd:   'phero-server serve -h' 1`] = `
{
  "command": "serve",
  "name": "help",
}
`;

exports[`parseServerCommand serve cmd:   'phero-server serve -p 1234 --verbose' 1`] = `
{
  "name": "serve",
  "port": 1234,
  "verbose": true,
}
`;

exports[`parseServerCommand serve cmd:   'phero-server serve -p 1234' 1`] = `
{
  "name": "serve",
  "port": 1234,
  "verbose": false,
}
`;

exports[`parseServerCommand serve cmd:   'phero-server serve' 1`] = `
{
  "name": "serve",
  "port": 3030,
  "verbose": false,
}
`;
